File: apps/patter-voice-sdk/libraries/python/getpatter/stt/openai_transcribe.py
"""OpenAI GPT-4o Transcribe STT for Patter pipeline mode."""

from __future__ import annotations

import os
from typing import ClassVar

from getpatter.providers.openai_transcribe_stt import (
    OpenAITranscribeSTT as _OpenAITranscribeSTT,
)

__all__ = ["STT"]


class STT(_OpenAITranscribeSTT):
    """OpenAI GPT-4o Transcribe STT (buffered HTTP transcription).

    ~10x faster than Whisper-1 — drop-in replacement for ``whisper.STT``.

    Example::

        from getpatter.stt import openai_transcribe

        stt = openai_transcribe.STT()                 # reads OPENAI_API_KEY
        stt = openai_transcribe.STT(api_key="sk-...", language="it")
    """

    provider_key: ClassVar[str] = "openai_transcribe"

    def __init__(
        self,
        api_key: str | None = None,
        *,
        language: str = "en",
        model: str = "gpt-4o-transcribe",
    ) -> None:
        key = api_key or os.environ.get("OPENAI_API_KEY")
        if not key:
            raise ValueError(
                "OpenAI Transcribe STT requires an api_key. Pass api_key='sk-...' or "
                "set OPENAI_API_KEY in the environment."
            )
        super().__init__(api_key=key, language=language, model=model)

File: apps/patter-voice-sdk/libraries/typescript/src/stt/openai-transcribe.ts
/** OpenAI GPT-4o Transcribe STT for Patter pipeline mode. */
import { OpenAITranscribeSTT as _OpenAITranscribeSTT } from "../providers/openai-transcribe-stt";
import type { WhisperResponseFormat } from "../providers/whisper-stt";

/** Constructor options for the OpenAI Transcribe `STT` adapter. */
export interface OpenAITranscribeSTTOptions {
  /** API key. Falls back to OPENAI_API_KEY env var when omitted. */
  apiKey?: string;
  /** ``gpt-4o-transcribe`` (default) or ``gpt-4o-mini-transcribe``. */
  model?: string;
  language?: string;
  bufferSize?: number;
  /** ``"json"`` only — the gpt-4o transcribe models reject ``"verbose_json"`` (whisper-1/WhisperSTT supports it). */
  responseFormat?: WhisperResponseFormat;
}

/**
 * OpenAI GPT-4o Transcribe STT — ~10x faster than Whisper-1.
 *
 * Drop-in replacement for ``whisper.STT`` with stronger multilingual
 * quality and significantly lower latency.
 *
 * @example
 * ```ts
 * import * as openaiTranscribe from "getpatter/stt/openai-transcribe";
 * const stt = new openaiTranscribe.STT();              // reads OPENAI_API_KEY
 * const stt = new openaiTranscribe.STT({ apiKey: "sk-...", language: "en" });
 * ```
 */
export class STT extends _OpenAITranscribeSTT {
  static readonly providerKey = "openai_transcribe";
  constructor(opts: OpenAITranscribeSTTOptions = {}) {
    const key = opts.apiKey ?? process.env.OPENAI_API_KEY;
    if (!key) {
      throw new Error(
        "OpenAI Transcribe STT requires an apiKey. Pass { apiKey: 'sk-...' } or " +
          "set OPENAI_API_KEY in the environment.",
      );
    }
    super(key, opts.language, opts.model ?? "gpt-4o-transcribe", opts.bufferSize, opts.responseFormat ?? "json");
  }
}
